src/client/dfs/dfs.c

```diff
@@ -1078,6 +1081,10 @@ entry_stat(dfs_t *dfs, daos_handle_t th, daos_handle_t oh, const char *name, siz
 			}
 
 			rc = daos_array_close(file_oh, NULL);
+			/* Try again in case of out of memory. Needed to avoid memory leak in
+			 * NLT fault injection */
+			if (rc == -DER_NOMEM)
+				rc = daos_array_close(file_oh, NULL);
```

Comment:
This again is an example of a bug in dfs that ideally we'd fix.  I'm not keen on this style of hack but there is a bug here that we should look at.

Reply:
https://github.com/daos-stack/daos/blob/master/src/client/api/array.c#L118-L120
When the fault injection causes dc_task_create() returns -DER_NOMEM, dc_task_schedule() is not called. The leaked memory was allocated by "D_ALLOC_PTR(array);" at line 91 in array_alloc() in dc_array.c,
https://github.com/daos-stack/daos/blob/master/src/client/array/dc_array.c#L91

Shall we retry "dc_task_create(dc_array_close, NULL, ev, &task);" in case it returns -DER_NOMEM inside daos_array_close()?

Reply:
If we retry "dc_task_create(dc_array_close, NULL, ev, &task);" in case it returns -DER_NOMEM inside daos_array_close(), we need to retry "dc_task_schedule(task, true)" too. How do you want to proceed? Thank you!

Reply:
ok. What should we do about the caused memory leak due to fault injection?

Reply:
> such hacks are really getting out of hand in the code. let's not add more please.

I agree, however this is a bug and it's impossible to address it at a higher level than dfs as the context has been lost by that point.  Outside the scope of this PR however.